src/v2/view-builder/views/phone/ChallengeAuthenticatorDataPhoneView.js

```diff
@@ -39,9 +39,17 @@ const Body = BaseForm.extend(
       const extraCssClasses =
         this.model.get('phoneNumber') !== loc('oie.phone.alternate.title', 'login') ?
           'strong no-translate' : '';
+      let nicknameText = '';
+      if (this.model.get('nickname')) {
+        nicknameText = '(' + this.model.escape('nickname') + ')';
+      }
+      if (nicknameText.length > 20) {
+        nicknameText = nicknameText.substring(0, 20) + '...';
+      }
```

Comment:
It is not really recommended to append to text string this way, specifically because of localization, in other languages, the way this is ordered may not make sense, but, we usually try to keep one translation key, that can be interpolated with parameter values. 

In this specific scenario, I would actually recommend creating new translation keys to handle the expected output. 

```
oie.phone.verify.sms.sendText.nickname = Send a code via SMS to <$1>{0}</$1> ({1})
oie.phone.verify.call.sendText.nickname = Send a code via voice call to <$1>{0}</$1> ({1})
```

Then to use it, you will have to just check for the existence of the `nickname` value, and if set and if phoneNumber is set, use one of the above keys and pass the values as parameters to the `loc` function. 

We can assist with this if you need help. However, if you go with this approach, you will have to create the translation keys and get them translated. 

cc: @jaredperreault-okta @shawyu-okta

Reply:
The nicknames should not be translated because those would not be Okta constants and will instead of be user defined strings.